### PR TITLE
Add request validation to controller submit

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Fixes https://github.com/giantswarm/inago/issues/66

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
RFR @giantswarm/cena

File: cli/submit.go [hunk 1]
```diff
@@ -23,6 +25,12 @@ func submitRun(cmd *cobra.Command, args []string) {
 		os.Exit(1)
 	}
 
+	ok, err := validator.ValidateRequest(req)
```
Comment (zeisss) [inline cli/submit.go]: don't we want this in the controller? So the controller validates it's input?

Comment (xh3b4sd) [inline cli/submit.go]: +1

File: controller/controller.go [hunk 2]
```diff
@@ -113,65 +82,10 @@ type controller struct {
...
-	}
-
-	return newRequest, nil
-}
-
-func (c controller) Submit(req Request) (*task.Task, error) {
-	if len(req.Units) == 0 {
-		return nil, maskAnyf(invalidArgumentError, "units must not be empty")
+func (c controller) Submit(req api.Request) (*task.Task, error) {
+	ok, err := validator.ValidateRequest(req)
```
Comment (JosephSalisbury) [inline controller/controller.go:86]: This is the new validation

**Comment by JosephSalisbury** [comment]
@giantswarm/cena RFR

I've had to pull out a bunch of stuff to get around the circular import _yay_

File: controller/controller.go (hunk 2 above)
Comment (xh3b4sd) [inline controller/controller.go:86]: Why is the validator not simply part of the controller? Then we can prevent all this restructuring. Looks like validation is a controller capability. Just put the source of the validation package into the controller package. What do you think?

Comment (JosephSalisbury) [inline controller/controller.go:86]: Yeah, actually, that makes way more sense. I'll scrap this PR, go with that.

**Pull request closed without merging**